qrenderdoc/Windows/MainWindow.cpp

```diff
@@ -1408,8 +1408,9 @@ void MainWindow::on_action_Attach_to_Running_Instance_triggered()
 
 void MainWindow::on_action_Manage_Remote_Servers_triggered()
 {
-  // the manager deletes itself when all lookups terminate
-  RDDialog::show(new RemoteManager(m_Ctx, this));
+  RemoteManager *dialog = new RemoteManager(m_Ctx, this);
+  RDDialog::show(dialog);
+  delete dialog;
```

Comment:
Ah right, if the lookups are already complete when you close the dialog it deletes itself immediately, which will screw up with further event processing.

What if you reverted the changes and instead replaced the `delete this;` with `deleteLater();` to delete it at a safer point back in the main event loop?